Source/Module/Login/LoginViewModel.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  LoginViewModel.swift
+//  GithubNotifications
+//
+//  Created by Chananchida F. on 9/13/21.
+//  Copyright © 2021 Nimblehq. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+import Combine
+
+protocol LoginViewModelProtocol {
+
+    func getUser(token: String, completion: @escaping CompletionHandler)
+}
+
+final class LoginViewModel: ObservableObject, LoginViewModelProtocol {
+
+    private var userService: UserServiceProtocol
+
+    var userSubscription: AnyCancellable?
+
+    init(userService: UserServiceProtocol) {
+        self.userService = userService
+    }
+    
+    func getUser(token: String, completion: @escaping CompletionHandler) {
+        UserSessionStorage.shared.authToken = token
+        userSubscription = userService.getUser()
```

Comment:
```suggestion
    init(userService: UserServiceProtocol, userSession: UserSessionStorage) {
        self.userService = userService
        self.userSession = userSession
    }
    
    func getUser(token: String, completion: @escaping CompletionHandler) {
        userSession.authToken = token
        userSubscription = userService.getUser()
```

Reply:
We should not use singleton `UserSessionStorage.shared`, the code cannot test with that :D